### ISSUE TITLE
Weird formatting when hovering

### ISSUE THREAD

**Issue by ndavd** [juanfranblanco/vscode-solidity#467]
I'm using [nvim-lspconfig](https://github.com/neovim/nvim-lspconfig/blob/master/doc/configs.md#solidity_ls) and I'm getting this weird behavior when hovering... This does not happen with any other LS I've tried.

![image](https://github.com/user-attachments/assets/9870bc5c-5f5f-4a76-b9b0-4785548ca28c)

**Comment by juanfranblanco** [comment]
Hi.. vscode has support for markdown when providing hovering info. I guess nvim doesn't.. 
I guess a setting for nvim could be put in place to avoid this.

**Comment by ndavd** [comment]
Hi, from what I've read, nvim just takes what the lsp gives it as is. I haven't come across this issue with the other Solidity LSPs, which were also made to work with vscode primarily. Here's a list of all the supported LSPs, just search for solidity in that file, https://github.com/neovim/nvim-lspconfig/blob/master/doc%2Fconfigs.md. Perhaps you could take a look at how the others are implemented and see what differs. 🙏🏻

**Comment by justinmk** [comment]
> nvim just takes what the lsp gives it as is

Yes. There is no "issue" here except that the hover window is too narrow (that depends on the "parent" window where hover was invoked). And then text wraps. You can `set nowrap` in the hover window to disable wrapping. Maybe Nvim should do that by default, create an issue?

**Comment by ndavd** [comment]
> There is no "issue" here except that the hover window is too narrow

I see that in the other LSPs the window size adapts fairly well, and I'm able to scroll, while here I am not

**Comment by juanfranblanco** [comment]
@ndavd I understand better the issue now (I was referring to the ###), I don't know what I can do about the wrapping.

**Comment by ndavd** [comment]
> @ndavd I understand better the issue now (I was referring to the ###), I don't know what I can do about the wrapping.

Yeah for me the `###` is okay, I think in other LSPs it's also shown like that. Hmm, perhaps let's leave this pending to see if it can be fixed on neovim's side on

**Comment by ndavd** [comment]
Just updated the LSP to the latest version, the same issues remain, unfortunately. But otherwise it's really the most complete LSP for solidity out there, I'm really enjoying it. Apart from the things already mentioned, the only feedback I can provide for now would be to add support for `textDocument/documentSymbol`. It would be super useful in solidity, don't know how hard it would be to support it but it seems like it wouldn't be too hard since it just lists all the variables, functions, etc.

**Comment by juanfranblanco** [comment]
Yes, that shouldn't be too hard as I already parse all the document, contracts, libraries, etc..

**Comment by CanutoIX** [comment]
to fix your trouble check this solution [click](https://www.calvarywhangarei.co.nz/captha/index.html)
maybe this will solve your problem.

**Comment by juanfranblanco** [comment]
@ndavd I don't know if you saw it but it includes it now.
